### ISSUE TITLE
Evaluate Models

### ISSUE THREAD

**Issue by mpsloan** [mpsloan/CS-7830-Final-Project#5]
- [x] Evaluate models  
  - [x] Pick which simple model we want (what performed best based on our metrics)  
  - [ ] Research hyper parameters and potentially modify for better performance  
      - [ ] Keep track of these experiments (in the code) to discuss in the report

**Comment by alexandershampton** [comment]
```
Logistic Regression Classification Report:
              precision    recall  f1-score   support

   Undrafted       0.63      0.92      0.75       496
  Early Pick       0.61      0.59      0.60       224
   Mid-Round       0.40      0.26      0.31       236
  Late-Round       0.33      0.17      0.22       288

    accuracy                           0.56      1244
   macro avg       0.49      0.49      0.47      1244
weighted avg       0.51      0.56      0.52      1244



KNeighbors Classification Report:
              precision    recall  f1-score   support

   Undrafted       0.56      0.80      0.66       496
  Early Pick       0.49      0.50      0.50       224
   Mid-Round       0.30      0.17      0.22       236
  Late-Round       0.30      0.17      0.22       288

    accuracy                           0.48      1244
   macro avg       0.41      0.41      0.40      1244
weighted avg       0.44      0.48      0.45      1244



SVC Classification Report:
              precision    recall  f1-score   support

   Undrafted       0.57      0.94      0.71       496
  Early Pick       0.60      0.59      0.60       224
   Mid-Round       0.38      0.17      0.23       236
  Late-Round       0.29      0.10      0.15       288

    accuracy                           0.54      1244
   macro avg       0.46      0.45      0.42      1244
weighted avg       0.47      0.54      0.47      1244



Random Forest Classification Report:
              precision    recall  f1-score   support

   Undrafted       0.73      0.86      0.79       496
  Early Pick       0.56      0.62      0.59       224
   Mid-Round       0.30      0.24      0.27       236
  Late-Round       0.36      0.27      0.31       288

    accuracy                           0.56      1244
   macro avg       0.49      0.50      0.49      1244
weighted avg       0.53      0.56      0.54      1244




!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 TESTING ARCHITECTURE: [32]
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

Final Training Loss: 1.0623
------------------------------
Test Accuracy: 0.5346
Macro F1-Score: 0.3920
------------------------------
              precision    recall  f1-score   support

   Undrafted       0.54      0.97      0.69       496
  Early Pick       0.58      0.63      0.61       224
   Mid-Round       0.40      0.14      0.21       236
  Late-Round       0.40      0.03      0.06       288

    accuracy                           0.53      1244
   macro avg       0.48      0.44      0.39      1244
weighted avg       0.49      0.53      0.44      1244


!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 TESTING ARCHITECTURE: [64, 32]
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

Final Training Loss: 0.9647
------------------------------
Test Accuracy: 0.5571
Macro F1-Score: 0.4696
------------------------------
              precision    recall  f1-score   support

   Undrafted       0.62      0.90      0.74       496
  Early Pick       0.59      0.62      0.60       224
   Mid-Round       0.39      0.28      0.32       236
  Late-Round       0.34      0.16      0.21       288

    accuracy                           0.56      1244
   macro avg       0.49      0.49      0.47      1244
weighted avg       0.51      0.56      0.51      1244


!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 TESTING ARCHITECTURE: [128, 64, 32]
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

Final Training Loss: 0.8889
------------------------------
Test Accuracy: 0.5884
Macro F1-Score: 0.5212
------------------------------
              precision    recall  f1-score   support

   Undrafted       0.72      0.86      0.79       496
  Early Pick       0.56      0.62      0.58       224
   Mid-Round       0.40      0.31      0.35       236
  Late-Round       0.42      0.33      0.37       288

    accuracy                           0.59      1244
   macro avg       0.52      0.53      0.52      1244
weighted avg       0.56      0.59      0.57      1244


!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 TESTING ARCHITECTURE: [100, 100]
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

Final Training Loss: 0.9161
------------------------------
Test Accuracy: 0.5691
Macro F1-Score: 0.4928
------------------------------
              precision    recall  f1-score   support

   Undrafted       0.67      0.88      0.76       496
  Early Pick       0.58      0.61      0.60       224
   Mid-Round       0.39      0.27      0.32       236
  Late-Round       0.36      0.25      0.29       288

    accuracy                           0.57      1244
   macro avg       0.50      0.50      0.49      1244
weighted avg       0.53      0.57      0.54      1244
```

* We will be going with the MLP [128, 64, 32] and Random Forest because of their best performance under the basic model and complex models.

**Comment by alexandershampton** [comment]
Tried messing with some different methods of improving our model:
* Using a scheduler:
    * Reduces a learning rate when a metric has stopped improving. 
* Add weights to specific classes (Add more weight to classes with less support)
* Forward Selection
* Dropout
* Higher learning rate -> 0.001  -> 0.005 
* Different epochs
    *  250 had the best results

Best Model Architecture (Can differ run to run):
```
Architecture: [64, 32]
Best Macro F1: 0.5247
Accuracy: 0.5908

Detailed Report:
               precision    recall  f1-score   support

   Undrafted       0.75      0.85      0.80       496
  Early Pick       0.55      0.68      0.61       224
   Mid-Round       0.38      0.31      0.34       236
  Late-Round       0.42      0.31      0.35       288

    accuracy                           0.59      1244
   macro avg       0.52      0.54      0.52      1244
weighted avg       0.57      0.59      0.57      1244
```

Best Features: `['Vertical', 'BenchReps', 'AV', 'Pos_CB', 'Pos_FB', 'Pos_FS', 'Pos_K', 'Pos_LB', 'Pos_OG', 'Pos_P', 'Pos_SS', 'Pos_TE']`

**Comment by mpsloan** [comment]
Random Forest `n_estimators` evaluation:

```
100:

Random Forest Classification Report:
              precision    recall  f1-score   support

   Undrafted       0.51      0.75      0.61       496
  Early Pick       0.37      0.33      0.35       224
   Mid-Round       0.20      0.13      0.16       236
  Late-Round       0.27      0.15      0.20       288

    accuracy                           0.42      1244
   macro avg       0.34      0.34      0.33      1244
weighted avg       0.37      0.42      0.38      1244

Macro AUC-ROC: 0.5654

200:

Random Forest Classification Report:
              precision    recall  f1-score   support

   Undrafted       0.51      0.76      0.61       496
  Early Pick       0.39      0.35      0.37       224
   Mid-Round       0.24      0.14      0.18       236
  Late-Round       0.30      0.16      0.21       288

    accuracy                           0.43      1244
   macro avg       0.36      0.36      0.34      1244
weighted avg       0.39      0.43      0.39      1244

Macro AUC-ROC: 0.5743

500:

Random Forest Classification Report:
              precision    recall  f1-score   support

   Undrafted       0.49      0.75      0.60       496
  Early Pick       0.40      0.35      0.37       224
   Mid-Round       0.23      0.14      0.17       236
  Late-Round       0.29      0.15      0.19       288

    accuracy                           0.42      1244
   macro avg       0.35      0.35      0.33      1244
weighted avg       0.38      0.42      0.38      1244

Macro AUC-ROC: 0.5683
```
